grakn-graql/src/main/java/ai/grakn/graql/internal/gremlin/EquivalentFragmentSet.java

```diff
@@ -38,45 +39,21 @@
  */
 public abstract class EquivalentFragmentSet implements Streamable<Fragment> {
 
-    private final ImmutableSet<Fragment> fragments;
-
-    /**
-     * @param fragments an array of Fragments that this EquivalentFragmentSet contains
-     */
-    protected EquivalentFragmentSet(Fragment... fragments) {
-        this.fragments = ImmutableSet.copyOf(fragments);
-    }
-
     /**
      * @return a set of fragments that this EquivalentFragmentSet contains
      */
-    public final Set<Fragment> fragments() {
-        return fragments;
-    }
-
-    @Override
-    public final boolean equals(Object o) {
-        if (this == o) return true;
-        if (o == null || getClass() != o.getClass()) return false;
-
-        EquivalentFragmentSet that = (EquivalentFragmentSet) o;
+    public abstract Set<Fragment> fragments();
 
-        return fragments != null ? fragments.equals(that.fragments) : that.fragments == null;
-
-    }
-
-    @Override
-    public final int hashCode() {
-        return fragments != null ? fragments.hashCode() : 0;
-    }
+    @Nullable
```

Comment:
I am starting to prefer Optionals over Nullables. . .  Just voicing a controversial opinion :)